src/common/xo/choose-sr-for-each-vdis-modal/index.js

```diff
@@ -32,80 +27,29 @@ Collapsible.propTypes = {
 }
 
 @propTypes({
-  vdis: propTypes.array.isRequired,
-  predicate: propTypes.func
+  mainSr: propTypes.string.isRequired,
+  mainSrPredicate: propTypes.func,
+  mapVdisSrs: propTypes.object.isRequired,
+  onChange: propTypes.func.isRequired,
+  srPredicate: propTypes.func,
+  vdis: propTypes.array.isRequired
 })
 export default class ChooseSrForEachVdisModal extends Component {
-  state = {
-    mapVdisSrs: {}
-  }
-
-  componentWillReceiveProps (newProps) {
-    if (
-      this.props.predicate !== undefined &&
-      newProps.predicate !== this.props.predicate
-    ) {
-      this.state = {
-        mainSr: undefined,
-        mapVdisSrs: {}
-      }
-    }
-  }
-
-  _onChange = props => {
-    this.setState(props)
-    this.props.onChange(props)
-  }
-
-  _onChangeMainSr = newSr => {
-    const oldSr = this.state.mainSr
-
-    if (oldSr == null || newSr == null || oldSr.$pool !== newSr.$pool) {
-      this.setState({
-        mapVdisSrs: {}
-      })
-    } else if (!newSr.shared) {
-      const mapVdisSrs = {...this.state.mapVdisSrs}
-      forEach(mapVdisSrs, (sr, vdi) => {
-        if (sr != null && newSr !== sr && sr.$container !== newSr.$container && !sr.shared) {
-          delete mapVdisSrs[vdi]
-        }
-      })
-      this._onChange({mapVdisSrs})
-    }
-
-    this._onChange({
-      mainSr: newSr
-    })
-  }
-
-  _getSrPredicate = createSelector(
-    () => this.state.mainSr,
-    () => this.state.mapVdisSrs,
-    (mainSr, mapVdisSrs) => sr =>
-      isSrWritable(sr) &&
-      mainSr.$pool === sr.$pool &&
-      areSrsCompatible(mainSr, sr) &&
-      every(mapVdisSrs, selectedSr => selectedSr == null || areSrsCompatible(selectedSr, sr))
-  )
-
   render () {
-    const { props, state } = this
-    const { vdis } = props
+    const { props } = this
     const {
       mainSr,
-      mapVdisSrs
-    } = state
-
-    const srPredicate = props.predicate || this._getSrPredicate()
+      mainSrPredicate = isSrWritable,
+      mapVdisSrs,
+      onChange,
+      srPredicate = mainSrPredicate,
+      vdis
+    } = props
```

Comment:
You can inline it if it's not used anywhere else.

I usually like to have my vars contextualized (`props.foo`, `state.bar`) instead of doing a big destructuring at the top of the method, but the choice is up to you.